pmd-core/src/main/java/net/sourceforge/pmd/PMDConfiguration.java

```diff
@@ -448,75 +469,162 @@ public List<String> getAllInputPaths() {
      */
     @Deprecated
     public void setInputPaths(String inputPaths) {
-        List<String> paths = new ArrayList<>();
-        Collections.addAll(paths, inputPaths.split(","));
+        List<Path> paths = new ArrayList<>();
+        for (String s : inputPaths.split(",")) {
+            paths.add(Paths.get(s));
+        }
         this.inputPaths = paths;
     }
 
     /**
      * Set the input paths to the given list of paths.
      * @throws NullPointerException If the parameter is null
+     * @deprecated Use {@link #setInputPathList(List)}
      */
+    @Deprecated
     public void setInputPaths(List<String> inputPaths) {
+        final List<Path> paths = new ArrayList<>(inputPaths.size());
+        for (String s : inputPaths) {
+            paths.add(Paths.get(s));
+        }
+        this.inputPaths = paths;
+    }
+
+    /**
+     * Set the input paths to the given list of paths.
+     * @throws NullPointerException If the parameter is null
+     */
+    public void setInputPathList(final List<Path> inputPaths) {
         this.inputPaths = new ArrayList<>(inputPaths);
     }
 
     /**
      * Add an input path. It is not split on commas.
      *
      * @throws NullPointerException If the parameter is null
+     * @deprecated Use {@link #addInputPath(Path)}
      */
+    @Deprecated
     public void addInputPath(String inputPath) {
+        Objects.requireNonNull(inputPath);
+        this.inputPaths.add(Paths.get(inputPath));
+    }
+
+    /**
+     * Add an input path. It is not split on commas.
+     *
+     * @throws NullPointerException If the parameter is null
+     */
+    public void addInputPath(Path inputPath) {
         Objects.requireNonNull(inputPath);
         this.inputPaths.add(inputPath);
     }
 
+    /**
+     * @deprecated Use {@link #getInputFile()}
+     */
+    @Deprecated
     public String getInputFilePath() {
+        return inputFilePath == null ? null : inputFilePath.toString();
+    }
+
+    public Path getInputFile() {
```

Comment:
Similarly, this is very vague and confusing given we have getInputPathList and other similar methods. I would name this `getInputFileListPath` maybe